platform/ios/src/MGLMapView.mm

```diff
@@ -1716,6 +1724,19 @@ - (void)handleRotateGesture:(UIRotationGestureRecognizer *)rotate
 
     self.cameraChangeReasonBitmask |= MGLCameraChangeReasonGestureRotate;
 
+    // Check whether a zoom triggered by a pinch gesture is occurring and if the rotation threshold has been met.
+    if (MGLDegreesFromRadians(self.rotationBeforeThresholdMet) < self.rotationThresholdWhileZooming && self.isZooming) {
+
+        self.rotationBeforeThresholdMet += abs(rotate.rotation);
+        rotate.delaysTouchesBegan = YES;
+        rotate.rotation = 0;
+
+        // If the gesture's state is `UIGestureRecognizerStateBegan`, continue. This prevents an exception where _changeDelimiterSuppressionDepth is less than 0.
+        if (rotate.state != UIGestureRecognizerStateBegan) {
```

Comment:
I do not believe that this fully solves the problem — after all, a rotation gesture may never actually happen, and yet this would allow the user tracking mode to change and for `notifyGestureDidBegin` and `notifyGestureDidEndWithDrift:` to be called.

Reply:
Even if the threshold is not met, once the gesture ends, this line will be hit: 
https://github.com/mapbox/mapbox-gl-native/pull/14929/files/4f4f72234f3b8afe73b6301ddd54edff335a8c57#diff-20502b6bc02986a4398d0da84b2d4b94R1812

The code within the check should only be hit if the user is pinch zooming. As far as I know, this means that the user tracking mode will already be changed before reaching this point.